core/src/test/scala/unit/kafka/server/ReplicaFetcherTierStateMachineTest.scala

```diff
@@ -121,4 +184,106 @@ class ReplicaFetcherTierStateMachineTest {
     assertTrue(fetcher.fetchState(partition).isEmpty)
     assertTrue(failedPartitions.contains(partition))
   }
+
+  private def mockBuildRemoteLogAuxState(mockReplicaMgr: ReplicaManager, topicPartition: TopicPartition): Unit = {
+    val idPartition = new TopicIdPartition(Uuid.randomUuid(), topicPartition)
+    logDir = JTestUtils.tempDirectory(s"kafka-${this.getClass.getSimpleName}")
+    val tpDir = JTestUtils.tempDirectory(logDir.toPath, idPartition.toString)
+    stateManager = new ProducerStateManager(topicPartition, tpDir, 5 * 60 * 1000, producerStateManagerConfig, time)
+    val unifiedLog = buildUnifiedLog(topicPartition, stateManager, tpDir)
+
+    doReturn(unifiedLog, Nil: _*).when(mockReplicaMgr).localLogOrException(any(classOf[TopicPartition]))
+    val mockRemoteLogManager = mock(classOf[RemoteLogManager])
+    doReturn(Option.apply(mockRemoteLogManager), Nil: _*).when(mockReplicaMgr).remoteLogManager
+
+    val remoteLogSegmentMetadata = new RemoteLogSegmentMetadata(
+      RemoteLogSegmentId.generateNew(new TopicIdPartition(Uuid.randomUuid(), topicPartition)),
+      4L, 4L, -1L, brokerId, -1L, 1024,
+      Optional.empty(), RemoteLogSegmentState.COPY_SEGMENT_FINISHED, Collections.singletonMap(4, 4L))
+    doReturn(Optional.of(remoteLogSegmentMetadata), Nil: _*).when(mockRemoteLogManager).fetchRemoteLogSegmentMetadata(any(classOf[TopicPartition]), ArgumentMatchers.eq(4), ArgumentMatchers.eq(4L))
```

Comment:
Do you not know the topicPartition which will be used here? Is there a reason to use matchers?